geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorMembershipListenerImpl.java

```diff
@@ -36,25 +38,30 @@
  *
  */
 public class LocatorMembershipListenerImpl implements LocatorMembershipListener {
-
-  private ConcurrentMap<Integer, Set<DistributionLocatorId>> allLocatorsInfo =
-      new ConcurrentHashMap<Integer, Set<DistributionLocatorId>>();
-
-  private ConcurrentMap<Integer, Set<String>> allServerLocatorsInfo =
-      new ConcurrentHashMap<Integer, Set<String>>();
-
   private static final Logger logger = LogService.getLogger();
-
-  private DistributionConfig config;
-
-  private TcpClient tcpClient;
+  static final String WAN_LOCATORS_DISTRIBUTOR_THREAD_NAME = "LocatorsDistributorThread";
+  static final int WAN_LOCATOR_DISTRIBUTION_RETRY_ATTEMPTS =
+      Integer.getInteger("WANLocator.LOCATOR_DISTRIBUTION_RETRY_ATTEMPTS", 3);
+  private static final String LISTENER_FAILURE_MESSAGE =
+      "Locator Membership listener could not exchange locator information {}:{} with {}:{} after {} retry attempts";
+  private static final String LISTENER_FINAL_FAILURE_MESSAGE =
+      "Locator Membership listener permanently failed to exchange locator information {}:{} with {}:{} after {} retry attempts";
 
   private int port;
+  private TcpClient tcpClient;
+  private DistributionConfig config;
+  private ConcurrentMap<Integer, Set<String>> allServerLocatorsInfo = new ConcurrentHashMap<>();
```

Comment:
Can tcpClient and these ConcurrentMaps be made `final`?

Reply:
Sure, thanks for pointing that out!.